### ISSUE TITLE
⌨ Applicant dashboard: "Your information" heading markup missing

### ISSUE THREAD

**Issue by MaddyDaigle** [GCTC-NTGC/gc-digital-talent#12886]
## ⌨ Accessibility Issue

Under the "dashboard" page, links under the “your information” section visually resemble headings (as links) but are not coded as such. This inconsistency makes navigation more difficult for screen reader users and keyboard users, as these elements are not recognized as structural landmarks. Additionally, this breaks consistency with the "applications and processes" and "career development" sections, where similar elements are marked as headings for easier navigation.

### WCAG 2.1 Criteria

WCAG 1.3.1 Info and Relationships – Information and structure must be conveyed through proper markup (e.g., using heading tags for elements that visually appear as headings).

WCAG 2.4.6 Headings and Labels – Headings and labels must describe the topic or purpose and be consistently used for better navigation.

Heuristic: Consistency and Standards – Users expect consistency in structure and navigation across similar sections.

## 🦋 Expected Behaviour

Links under “your information” that visually resemble headings should be marked up as headings (using appropriate tags such as h2, h3, etc. where relevant). This would ensure consistent navigation and better accessibility, aligning with other sections on the dashboard.

## 🕵️ Details

N/A

## 📋 Steps to Reproduce

1. Navigate to applicant dashboard by logging into the platform
2. Make sure the role is set to "Applicant"

## 📸 Screenshot

![Image](https://github.com/user-attachments/assets/16143008-5926-495f-bc79-26c706a00ff9)

## 🙋‍♀️ Proposed Solution

Wrap links in `Heading level=3`. Check to ensure proper rank but I think 3 is correct.

Since this is a pre-built component, you will probably need to add a prop with the heading level.

Maybe want to use the native element if the styling in our Heading component are much different.

## ✅ Acceptance Criteria

- [ ] Links are also semantic headings
- [ ] Appropriate heading rank is applied
- [ ] No styling changes, just semantic hierarchy changes
